### PR TITLE
Rachel Chitter Challenge

### PR DESCRIPTION
**Pull request by rachjgriff**

### CODE REVIEW

File: db/migrations/01_create_peeps_table.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+CREATE TABLE peep(id SERIAL PRIMARY KEY, peep VARCHAR(500), timestamp TIMESTAMPTZS DEFAULT NOW())
```
Comment (rhysco8) [inline db/migrations/01_create_peeps_table.sql]: I found a couple of typos in here
```suggestion
CREATE TABLE peep(id SERIAL PRIMARY KEY, peep VARCHAR(500), timestamp TIMESTAMPTZ DEFAULT NOW());
```

File: README.md [hunk 2]
```diff
@@ -1,4 +1,86 @@
...
+------------
+* Connect to psql
+* Follow the below instructions to create the required database.
+
+```
+CREATE DATABASE "Chitter";
+\c Chitter;
+```
+```
+CREATE DATABASE "Chitter_Test"
```
Comment (rhysco8) [inline README.md]: This is missing a `;` at the end of the line

File: README.md [hunk 3]
```diff
@@ -1,4 +1,86 @@
...
+* Log in to Chitter
+* Log out of Chitter
+* Tag a user in a Peep
+* Receive an email if you are tagged in a Peep
+
+### APPROACH
+------------
+* Diagramed MVC Model of Chitter
+* Diagramed Class Responsibility Collaborator (CRC) Model of Chitter: https://github.com/rachjgriff/chitter-challenge/blob/master/UNADJUSTEDNONRAW_thumb_d77.jpg
+* Created Database structure for the Domain Model using Entity Relationship diagram: https://github.com/rachjgriff/chitter-challenge/blob/master/UNADJUSTEDNONRAW_thumb_d69.jpg
```
Comment (rhysco8) [inline README.md]: I think you could have embedded this image and the one above rather than linking

Comment (rhysco8) [reply README.md]: Why did you choose to have the images in the root directory rather than an image folder?

File: Gemfile [hunk 4]
```diff
@@ -2,8 +2,13 @@ source 'https://rubygems.org'
 
 ruby '2.5.0'
 
+gem 'bcrypt'
+gem 'capybara'
```
Comment (rhysco8) [inline Gemfile]: `capybara` should be in your test group below, along with `rubocop`

File: spec/features/visiting_Chitter_test_spec.rb [hunk 5]
```diff
@@ -0,0 +1,7 @@
+feature 'Test' do
```
Comment (rhysco8) [inline spec/features/visiting_Chitter_test_spec.rb]: Why did you choose not to have any peeps displayed on this page for this specific feature test?

File: spec/features/new_post_spec.rb [hunk 6]
```diff
@@ -0,0 +1,10 @@
+feature "New Post" do
+  scenario "User able to add a new post to Chitter" do
+    visit("/peeps/new")
+
+    fill_in("peep", with: "Hi Chitter!")
+    click_button("Post")
+
+    expect(page).to have_content "Hi Chitter!"
```
Comment (rhysco8) [inline spec/features/new_post_spec.rb]: Are there some more expectations you could include in this feature test, like expecting the page to display the time the peep was made?

File: spec/features/sign_up_spec.rb [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+  scenario "A user can sign up and become a member of Chitter" do
+    visit("/members/new")
+
+    fill_in('name', with: 'Test Person')
+    fill_in('username', with: 'test123')
+    fill_in('email', with: 'test@example.com')
+    fill_in('password', with: 'password123')
+    click_button('Submit')
+
+    expect(page).to have_content "Welcome to Chitter, Test Person"
```
Comment (rhysco8) [inline spec/features/sign_up_spec.rb]: Could you write an expectation for the database to contain this user?

File: spec/features/new_post_spec.rb (hunk 6 above)
Comment (rhysco8) [reply spec/features/new_post_spec.rb]: Or for the peep db to contain this peep?

File: spec/features/authentication_spec.rb [hunk 8]
```diff
@@ -0,0 +1,66 @@
+feature 'authentication' do
```
Comment (rhysco8) [inline spec/features/authentication_spec.rb]: Nice! This is a thorough feature test of use and edge cases

File: spec/features/authentication_spec.rb [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+  scenario 'A Chitter member can log in' do
+    Member.create(
```
Comment (rhysco8) [inline spec/features/authentication_spec.rb]: Could you have kept your code DRY by putting this in a helper method?

File: app/app.rb [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+      redirect('/peeps')
+    else
+      flash[:notice] = 'Please check your email or password'
+      redirect('/sessions/new')
+    end
+  end
+
+  post '/sessions/destroy' do
+    session.clear
+    flash[:notice] = 'You have signed out'
```
Comment (rhysco8) [inline app/app.rb]: Could this have been `flash.next[:notice]` to see the message after the redirect?
